hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/file/TestOMRecoverLeaseRequest.java

```diff
@@ -79,26 +91,230 @@ public void testRecoverHsyncFile() throws Exception {
+    RecoverLeaseResponse recoverLeaseResponse = omResponse.getRecoverLeaseResponse();
+    KeyInfo keyInfo1 = recoverLeaseResponse.getKeyInfo();
+    Assertions.assertNotNull(keyInfo1);
+
+    // call recovery second time
+    omClientResponse = validateAndUpdateCache();
+    omResponse = omClientResponse.getOMResponse();
+    Assertions.assertEquals(OzoneManagerProtocolProtos.Status.OK, omResponse.getStatus());
+    recoverLeaseResponse = omResponse.getRecoverLeaseResponse();
+    KeyInfo keyInfo2 = recoverLeaseResponse.getKeyInfo();
+    Assertions.assertNotNull(keyInfo2);
+    Assertions.assertEquals(keyInfo1.getKeyName(), keyInfo2.getKeyName());
+  }
+
+  /**
+   * Verify that COMMIT request for recovery is not idempotent.
+   * @throws Exception
+   */
+  @Test
+  public void testCommitStageNotIdempotent() throws Exception {
+    when(ozoneManager.getAclsEnabled()).thenReturn(true);
+    when(ozoneManager.getVolumeOwner(
+        anyString(),
+        any(IAccessAuthorizer.ACLType.class), any(
+            OzoneObj.ResourceType.class)))
+        .thenReturn("user");
+    InetSocketAddress address = new InetSocketAddress("localhost", 10000);
+    when(ozoneManager.getOmRpcServerAddr()).thenReturn(address);
+    populateNamespace(true, true, true, true);
+
+    // call recovery
+    OMClientResponse omClientResponse = validateAndUpdateCache();
+    OMResponse omResponse = omClientResponse.getOMResponse();
+    Assertions.assertEquals(OzoneManagerProtocolProtos.Status.OK, omResponse.getStatus());
+    RecoverLeaseResponse recoverLeaseResponse = omResponse.getRecoverLeaseResponse();
+    KeyInfo keyInfo = recoverLeaseResponse.getKeyInfo();
+    Assertions.assertNotNull(keyInfo);
+    OmKeyInfo omKeyInfo = OmKeyInfo.getFromProtobuf(keyInfo);
+
+    KeyArgs newKeyArgs = getNewKeyArgs(omKeyInfo, 0);
+
+    // call commit first time
+    omClientResponse = validateAndUpdateCacheForCommit(newKeyArgs);
+    omResponse = omClientResponse.getOMResponse();
+    Assertions.assertEquals(OzoneManagerProtocolProtos.Status.OK, omResponse.getStatus());
+
+    // call commit second time
+    omClientResponse = validateAndUpdateCacheForCommit(newKeyArgs);
+    omResponse = omClientResponse.getOMResponse();
+    Assertions.assertEquals(OzoneManagerProtocolProtos.Status.KEY_ALREADY_CLOSED, omResponse.getStatus());
+  }
+
+  /**
+   * Verify that RecoverLease COMMIT request has a new file length.
+   * @throws Exception
+   */
+  @Test
+  public void testRecoverWithNewFileLength() throws Exception {
+    when(ozoneManager.getAclsEnabled()).thenReturn(true);
+    when(ozoneManager.getVolumeOwner(
+        anyString(),
+        any(IAccessAuthorizer.ACLType.class), any(
+            OzoneObj.ResourceType.class)))
+        .thenReturn("user");
+    InetSocketAddress address = new InetSocketAddress("localhost", 10000);
+    when(ozoneManager.getOmRpcServerAddr()).thenReturn(address);
```

Comment:
Above ACL and address usage seems to be in every new test method, may be we can move to init or other method and use.

Reply:
Actually not needed, removed them.